### PR TITLE
RESTEASY-2979 remove spring components

### PR DESCRIPTION
**Pull request by liweinan**

### CODE REVIEW

File: jboss-modules/build.xml [hunk 1]
```diff
@@ -182,11 +182,11 @@
         </module-def>
 
         <module-def name="org.jboss.resteasy.resteasy-spring">
-            <maven-resource-bundle group="org.jboss.resteasy" artifact="resteasy-spring" subdir="bundled/resteasy-spring-jar"/>
+            <maven-resource-bundle group="org.jboss.resteasy.spring" artifact="resteasy-spring" subdir="bundled/resteasy-spring-jar"/>
```
Comment (liweinan) [inline jboss-modules/build.xml:185]: This way we can workaround current jboss-module solution (using the new resteasy-spring components without changing anything in jboss-module descriptors.

Provement:

```bash
$ unzip -l resteasy-jboss-modules-5.0.0-SNAPSHOT.zip | grep spring                                                                                                                                                                                                                                              21:44:28
        0  07-27-2021 21:43   org/jboss/resteasy/resteasy-spring/
        0  07-27-2021 21:43   org/jboss/resteasy/resteasy-spring/main/
        0  07-27-2021 21:43   org/jboss/resteasy/resteasy-spring/main/bundled/
        0  07-27-2021 21:43   org/jboss/resteasy/resteasy-spring/main/bundled/resteasy-spring-jar/
        0  07-27-2021 21:43   org/jboss/resteasy/resteasy-spring-web/
        0  07-27-2021 21:43   org/jboss/resteasy/resteasy-spring-web/main/
    54992  07-27-2021 21:43   org/jboss/resteasy/resteasy-spring/main/bundled/resteasy-spring-jar/resteasy-spring-1.0.0.Final.jar
     1183  07-27-2021 21:43   org/jboss/resteasy/resteasy-spring/main/module.xml
    17952  07-27-2021 21:43   org/jboss/resteasy/resteasy-spring-web/main/resteasy-spring-web-1.0.0.Final.jar
     1184  07-27-2021 21:43   org/jboss/resteasy/resteasy-spring-web/main/module.xml
```

```bash
$ unzip -p resteasy-jboss-modules-5.0.0-SNAPSHOT.zip org/jboss/resteasy/resteasy-spring/main/module.xml | grep spring                                                                                                                                                                                           21:46:02
<module xmlns="urn:jboss:module:1.9" name="org.jboss.resteasy.resteasy-spring">
```

**Comment by liweinan** [comment]
Until now only these places have `spring` as dependencies:

```bash
$ grep -ril 'spring' *                                                                                                                                                                                                                                                                                          21:48:28
arquillian-utils/src/main/java/org/jboss/resteasy/utils/maven/MavenUtil.java
arquillian-utils/src/main/java/org/jboss/resteasy/utils/TestUtilSpring.java
integration-tests/pom.xml
```

The first two are actually doesn't use any `spring` or `resteasy-spring` things, so I guess in the future we can move these two classes to `resteasy-spring` project.

For the `spring` part in last `pom.xml`, it actually deals with resteasy spring modules, but as we haven't changed the structure of `jboss-modules`, so nothing needed to be changed here. (until we change everything to Galleon).

File: jboss-modules/build.xml (hunk 1 above)
Comment (jamezp) [reply jboss-modules/build.xml:185]: I think we just need to completely remove the modules and references to them in the main project.

File: jboss-modules/build.xml [hunk 2]
```diff
@@ -182,11 +182,11 @@
...
 
         <module-def name="org.jboss.resteasy.resteasy-spring">
-            <maven-resource-bundle group="org.jboss.resteasy" artifact="resteasy-spring" subdir="bundled/resteasy-spring-jar"/>
+            <maven-resource-bundle group="org.jboss.resteasy.spring" artifact="resteasy-spring" subdir="bundled/resteasy-spring-jar"/>
         </module-def>
 
         <module-def name="org.jboss.resteasy.resteasy-spring-web">
-            <maven-resource group="org.jboss.resteasy" artifact="resteasy-spring-web"/>
+            <maven-resource group="org.jboss.resteasy.spring" artifact="resteasy-spring-web"/>
         </module-def>
```
Comment (jamezp) [inline jboss-modules/build.xml:190]: This should just be deleted.

File: jboss-modules/pom.xml [hunk 3]
```diff
@@ -207,21 +302,43 @@
...
             <artifactId>resteasy-spring-web</artifactId>
-            <version>${project.version}</version>
+            <version>${version.org.jboss.resteasy.spring}</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.jboss.resteasy</groupId>
+                    <artifactId>resteasy-core</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
```
Comment (jamezp) [inline jboss-modules/pom.xml:330]: These should just be removed.

File: pom.xml [hunk 4]
```diff
@@ -24,6 +24,7 @@
         <dep.arquillian-bom.version>1.4.1.Final</dep.arquillian-bom.version>
         <version.org.jboss.arquillian.container.arquillian-weld-embedded>2.0.0.Final</version.org.jboss.arquillian.container.arquillian-weld-embedded>
         <version.org.jboss.resteasy.checkstyle>1.0.0.Final</version.org.jboss.resteasy.checkstyle>
+        <version.org.jboss.resteasy.spring>1.0.0.Final</version.org.jboss.resteasy.spring>
```
Comment (jamezp) [inline pom.xml:27]: We shouldn't add Spring to this as we end up with a circular dependency.

File: resteasy-bom/pom.xml [hunk 5]
```diff
@@ -77,14 +77,14 @@
                 <version>${project.version}</version>
             </dependency>
             <dependency>
-                <groupId>org.jboss.resteasy</groupId>
+                <groupId>org.jboss.resteasy.spring</groupId>
```
Comment (jamezp) [inline resteasy-bom/pom.xml:80]: Same with these, I think they should be removed.

**Review by jamezp — CHANGES_REQUESTED** [review]
I think all Spring related stuff, including stuff in the `arquillian-utils`, should be moved to the new resteasy-spring project. In other words, the top RESTEasy project should have no references to Spring.

**Comment by liweinan** [comment]
@jamezp Thanks for checking James! I'll make the change according to the reviews.

**Comment by liweinan** [comment]
@jamezp I have removed all the spring stuff, could you pls help to check? Thanks!

The next step I'll go back to `resteasy-spring` side to go on working on `Galleon` support.

**Review by jamezp — APPROVED** [review]

**Comment by liweinan** [comment]
Thanks for review James!

**Pull request merged**